### PR TITLE
Fix logic for classList methods

### PR DESCRIPTION
**Pull request by diervo**
The change commit 1e80f5082a2b2c497a3b3f6ff692c1e98cfdbe1b broke the logic on the classList methods. 

The tests are incomplete and only testing the DOMTokenList override, but not the whole classList prototype on Element.

Reverted the logic and added tests.

### CODE REVIEW

**Comment by diervo** [comment]
IMHO `~` its very hard to read and understand, so I regularly stay out of it.
My 2cents.

File: classList.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ classListProto.add = function () {
 	;
 	do {
 		token = tokens[i] + "";
-		if (~checkTokenAndGetIndex(this, token)) {
+		if (checkTokenAndGetIndex(this, token) === -1) {
```
Comment (eligrey) [inline classList.js]: This would be `!== -1`

Comment (diervo) [reply classList.js]: @eligrey No, if is == -1, does not exist, therefore we need to add it.
The test will fail if I change this

Comment (kevin01001101) [reply classList.js]: I believe @diervo is correct.  I just experienced this issue in a project where we're using this fine poly.
Yes, (~x)  == (x !== -1).

But I think some errors were introduced in a [previous commit](https://github.com/eligrey/classList.js/commit/1e80f5082a2b2c497a3b3f6ff692c1e98cfdbe1b), which is what @divero is pointing out.

Line 97 was originally changed  from x !== -1 to !~x.  Which I think should have been ~x.
Line 109 was changed from x === -1 to ~x.  Which I think should have been !~x.  
Line 132 changed from x !== -1 to ~x.  Which I think was proper.

Regardless of whether (in)equality or the complement operator are used, there is a logic error that should be corrected.

Comment (eligrey) [reply classList.js]: Addressing this now

**Pull request closed without merging**